src/main/java/appeng/util/InventoryAdaptor.java

```diff
@@ -49,60 +41,29 @@
  */
 public abstract class InventoryAdaptor implements Iterable<ItemSlot>
 {
-
-	// returns an appropriate adaptor, or null
-	public static InventoryAdaptor getAdaptor( final Object te, final EnumFacing d )
+	public static InventoryAdaptor getAdaptor( final TileEntity te, final EnumFacing d )
 	{
-		if( te == null )
-		{
-			return null;
-		}
-
-		if( te instanceof ICapabilityProvider )
+		if( te != null )
 		{
-			ICapabilityProvider capProvider = (ICapabilityProvider) te;
-
-			// Attempt getting an IItemHandler for the given side via caps
-			IItemHandler itemHandler = capProvider.getCapability( CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, d );
-			if( itemHandler != null && itemHandler.getSlots() > 0 )
+			if( te.hasCapability( CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, d ) )
 			{
-				return new AdaptorItemHandler( itemHandler );
+				// Attempt getting an IItemHandler for the given side via caps
+				IItemHandler itemHandler = te.getCapability( CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, d );
+				if( itemHandler != null )
+				{
+					return new AdaptorItemHandler( itemHandler );
+				}
 			}
 		}
+		return null;
+	}
 
-		if( te instanceof EntityPlayer )
-		{
-			return new AdaptorIInventory( new AdaptorPlayerInventory( ( (EntityPlayer) te ).inventory, false ) );
-		}
-		else if( te instanceof ArrayList )
-		{
-			@SuppressWarnings( "unchecked" )
-			final ArrayList<ItemStack> list = (ArrayList<ItemStack>) te;
-
-			return new AdaptorList( list );
-		}
-		else if( te instanceof TileEntityChest )
-		{
-			return new AdaptorIInventory( Platform.GetChestInv( te ) );
-		}
-		else if( te instanceof ISidedInventory )
-		{
-			final ISidedInventory si = (ISidedInventory) te;
-			final int[] slots = si.getSlotsForFace( d );
-			if( si.getSizeInventory() > 0 && slots != null && slots.length > 0 )
-			{
-				return new AdaptorIInventory( new WrapperMCISidedInventory( si, d ) );
-			}
-		}
-		else if( te instanceof IInventory )
+	public static InventoryAdaptor getAdaptor( final EntityPlayer te )
+	{
+		if( te != null )
 		{
-			final IInventory i = (IInventory) te;
-			if( i.getSizeInventory() > 0 )
-			{
-				return new AdaptorIInventory( i );
-			}
+			return new AdaptorItemHandler( new PlayerInvWrapper( ( (EntityPlayer) te ).inventory ) );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "EntityPlayer". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1905)